### DISCUSSION TITLE
How to handle html without React?

### DISCUSSION THREAD

**Discussion by V-Krause** [i18next/i18next#1716] [Q&A]
I am currently programming a desktop app in Electron. I have previously created html pages and implemented them in the Electron program. First I worked with React and convert my html files in .jsx files, which made the implementation of i18next easy. But since I had to realize that React is not suitable for my app, I now switched back to the basic structure of Electron. 

To implement i18next without React, I first tried i18next-electron-fs-backend, since it is supposed to be a serverless app. The problem now is that I don't know how to call i18next in the html files, because unfortunately there was no example available. I have also looked at other packages and solutions, but unfortunately none have a concrete html solution that worked for my case. My goal is to translate something like the labels of buttons or table contents, which are fixed, using defined JSON files that contain the translations.

My rough programm structure is:
|─html
____└─── mainwindow,html
____└─── settings.html
____└─── ...
|─js
____└─── localization
____________└─── locales
____________________└─── de
____________________________└─── translation.json
____________________└─── en
____________└─── i18next.config.js
____└─── main.js
____└─── preload.js
____└─── js

e.g. translation.json in de
`{
  "Hello": "Hallo",
  "File": "Datei",
  "Exit": "Ausgang",
  "Edit": "Bearbeiten"
}`

I have followed the steps of i18next-electron-fs-backend so far and implemented them.

My question now is, how can I customize for example the label of a button, so that I can translate it into the desired language by selecting a language (e.g. with the help of the titlebar, like in the secure-electron-template)?
Or is there another package from i18next, which I can use?

Thank you!
Vincent

**Comment by jamuhl** [comment]
plain html...use it like https://www.i18next.com/overview/getting-started#basic-sample

or use something like:

- https://github.com/i18next/jquery-i18next
- https://github.com/mthh/loc-i18next
- https://github.com/i18next/i18nextify
